src/main/java/org/codehaus/groovy/vmplugin/v9/Java9.java

```diff
@@ -106,6 +141,15 @@
         return result;
     }
 
+    @Override
+    public boolean resolveFromDefaultImports(ResolveVisitor resolveVisitor, final ClassNode type) {
```

Comment:
what is java9 specific about this code, that it has to go into a VM Plugin?

Reply:
The PR is based on the classes found by `ClassFinder`, which is based on Java 9+ API, so it has to go into a VM plugin.

Reply:
Actually, ClassFinder my have handling for module-info and jrt,  but it would work just fine under Java8, if you added handling for jar urls I think. Are you using methods/classes in there that are Java9 specific? Did not spot any, but I was not looking too much into details.
Anyway, my thinking is that the plugins should not just bring compiler classes into them if not needed. In fact we should move them out there as much as possible. The Java5 generics code was a bit different since you needed to produce AST and required Java5 specific classes to produce it. But now that is no longer the case and a good portion of the methods should be no longer in a plugin

Reply:
>  In fact we should move them out there as much as possible.

Understood. We use `ClassFinder` to find default imported classes within JRT as well, but visiting JRT is only available in Java 9+  :-(